Tests/CSharp/src/Tests/AssetSystemTests.cs

```diff
@@ -1380,6 +1380,110 @@ public static void thirdPartyPackagedAssetIdentifierTest()
+                using var result = assetSystem.UploadAssetData(sourceAssetCollection, sourceAsset, fileSource).Result;
+                var resCode = result.GetResultCode();
+
+                Assert.AreEqual(resCode, Systems.EResultCode.Success);
+            }
+
+            // Create 'dest' space and invoke the copy
+            var destSpace = SpaceSystemTests.CreateSpace(spaceSystem, destSpaceName, testSpaceDescription, Systems.SpaceAttributes.Private, null, null, null);
+            var destAssetCollections = new Common.Array<Systems.AssetCollection>();
+            {
+                Common.Array<Systems.AssetCollection> sourceAssetCollections = new Common.Array<Systems.AssetCollection>(1);
+                sourceAssetCollections[0] = sourceAssetCollection;
+
+                using var result = assetSystem.CopyAssetCollectionsToSpace(sourceAssetCollections, destSpace.Id, false).Result;
+                
+                Assert.AreEqual(result.GetResultCode(), Systems.EResultCode.Success);
+                destAssetCollections = result.GetAssetCollections();
+            }
+
+            // Validate the copied asset collection and its data
+	        {
+                Assert.AreEqual(destAssetCollections.Size(), 1UL);
+		        Assert.AreNotEqual(destAssetCollections[0].Id, sourceAssetCollection.Id);
+		        Assert.AreEqual(destAssetCollections[0].SpaceId, destSpace.Id);
+		        Assert.AreEqual(destAssetCollections[0].Type, sourceAssetCollection.Type);
+		        Assert.AreEqual(destAssetCollections[0].Tags.Size(), 1UL);
+
+                GetAssetsInCollection(assetSystem, destAssetCollections[0], out var destAssets);
+                Assert.AreEqual(destAssets.Size(), 1UL);
+
+                // Get the copied data and compare it with our source
+                using var result = assetSystem.DownloadAssetData(destAssets[0]).Result;
+                Assert.AreEqual(result.GetResultCode(), Systems.EResultCode.Success);
+
+                byte[] downloadedData = new byte[result.GetDataLength()];
+                Marshal.Copy(result.GetData(), downloadedData, 0, (int)result.GetDataLength());
+
+                var fileData = File.ReadAllBytes(Path.GetFullPath("assets/test.json"));
+                Assert.IsTrue(downloadedData.SequenceEqual(fileData));
+            }
+
+            // Validating that we must have at least one asset collection to copy
+	        {
+		        Common.Array<Systems.AssetCollection> sourceAssetCollections = new Common.Array<Systems.AssetCollection>(1);
+                using var result = assetSystem.CopyAssetCollectionsToSpace(sourceAssetCollections, destSpace.Id, false).Result;
+                Assert.AreEqual(result.GetResultCode(), Systems.EResultCode.Failed);
+	        }
+
+            // Validating we cannot perform a copy if the asset has no space ID
+            {
+                Common.Array<Systems.AssetCollection> assetCollections = new Common.Array<Systems.AssetCollection>(1);
```

Comment:
Same here as above in regards to `using var`.

Reply:
Addressed in latest